### PR TITLE
feat: Adding support for CLI

### PR DESCRIPTION
**Pull request by jdalrymple**
closes: #146

There may be a cleaner way of running the generate script and not having to have two tsconfigs, or use esm. Any suggestions are welcome :D

TODO: 
- [x] Review
- [x] Add improved config *Optional*
- [x] Add docs

### CODE REVIEW

**Comment by jdalrymple** [comment]
@casz Mind looking this over?

**Comment by casz** [comment]
@jdalrymple In Kiev at the moment but I'll will give it a look either Thursday or Friday if's that is okay :)

123 files 😓

**Comment by jdalrymple** [comment]
Just doing a quick look around, the creator of this package has a much nicer way of supporting config (the tokens, url etc). Probably should incorporate this sort of support

https://www.npmjs.com/package/git-lab-cli

**Comment by casz** [comment]
Reviewing 😓

File: src/bin/generate.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import getParamNames from 'get-param-names';
+import { outputJsonSync, removeSync } from 'fs-extra';
+import * as core from '../core';
+import { BaseService } from '../core/infrastructure';
```
[the remark this answers is not in the document]
Comment (jdalrymple) [reply src/bin/generate.ts]: the core doesnt export the BaseService :(

[the remark this answers is not in the document]
Comment (casz) [reply src/bin/generate.ts]: Than I was mistaken 😅

File: README.md [hunk 2]
```diff
@@ -41,7 +42,64 @@
...
+| `rejectUnauthorized` | Yes      | `false`                                               | Http Certificate setting                                        |
+| `sudo`               | Yes      | `false`                                               | Sudo query parameter                                            |
+| `version`            | Yes      | `v4`                                                  | API Version ID                                                  |
+| `camelize`           | Yes      | `false`                                               | Response Key Camelize. Camelizes all response body keys         |
+| `requester`          | Yes      | [KyRequester.ts](./src/infrastructure/KyRequester.ts) | Request Library Wrapper. Currently wraps Ky.                    |
+| `requestTimeout`     | Yes      | `300000`                                              | Request Library Timeout in ms                                   |
+
+### CLI Support
+
+The CLI export functions in a similar manner, following the pattern:
```
Comment (casz) [inline README.md]: Missing example of dotenv or passing env from command :)

Comment (jdalrymple) [reply README.md]: Done!

**Pull request merged**